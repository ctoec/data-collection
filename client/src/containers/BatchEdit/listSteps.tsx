import { StepProps } from '@ctoec/component-library';
import React from 'react';
import {
  RecordFormProps,
  formSections,
  SECTION_KEYS,
  ChildInfoForm,
  ChildIdentifiersForm,
  FamilyAddressForm,
} from '../../components/Forms';
import { Child } from '../../shared/models';
import { EnrollmentFundingForm } from './EnrollmentFunding/Form';

const forms = [
  { key: SECTION_KEYS.IDENT, form: ChildIdentifiersForm },
  { key: SECTION_KEYS.DEMO, form: ChildInfoForm },
  { key: SECTION_KEYS.FAMILY, form: FamilyAddressForm },
  { key: SECTION_KEYS.ENROLLMENT, form: EnrollmentFundingForm },
];

/**
 * Util that determines which forms have errors for a particular child
 * and creates a formated string to display in the BatchEdit cards
 * below the name
 */
export const formatBatchEditPlaceholderString = (child: Child) => {
  const formsWithErrors = formSections.filter((section) =>
    section.hasError(child)
  );
  let errorStr = formsWithErrors
    .map(({ key, name, hasError }) => name)
    .join(', ');
  return errorStr;
};

export const listSteps = (child: Child) =>
  formSections
    .map(({ key, name, hasError }) => {
      const Form = forms.find((s) => s.key === key)?.form || (() => <></>);
      if (hasError(child)) {
        return {
          key,
          name,
          status: (props) =>
            hasError(props.child) ? 'incomplete' : 'complete',
          Form,
        } as StepProps<RecordFormProps>;
      }
    })
    .filter((step) => !!step) as StepProps<RecordFormProps>[];
