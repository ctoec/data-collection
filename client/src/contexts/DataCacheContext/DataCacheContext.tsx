import React, { createContext, useState, useEffect, useContext } from 'react';
import { Child } from '../../shared/models';
import AuthenticationContext from '../AuthenticationContext/AuthenticationContext';
import { apiGet } from '../../utils/api';

type ReadOnlyDataCache<T> = {
  records: T[];
  loading: boolean;
};
type ReadWriteDataCache<T> = ReadOnlyDataCache<T> & {
  addOrUpdateRecord: (record: T) => void;
};

export type DataCacheContextType = {
  children: ReadWriteDataCache<Child>;
};

const DataCacheContext = createContext<DataCacheContextType>({
  children: { records: [], addOrUpdateRecord: (_) => {}, loading: true },
});

const { Provider, Consumer } = DataCacheContext;

const DataCacheProvider: React.FC = ({ children: childNodes }) => {
  // CHILD CACHE
  const [children, setChildren] = useState<Child[]>([]);
  const [childrenLoading, setChildrenLoading] = useState(false);
  const [haveFetchedChildren, setHaveFetchedChildren] = useState(false);
  const addOrUpdateChild = (_record: Child) => {
    setChildren((existingRecords) => [
      ...existingRecords.filter((r) => r.id !== _record.id),
      _record,
    ]);
  };
  const childCache = {
    records: children,
    loading: childrenLoading,
    addOrUpdateRecord: addOrUpdateChild,
  };
  // END CHILD CACHE

  const { accessToken } = useContext(AuthenticationContext);
  useEffect(() => {
    if (!haveFetchedChildren) {
      setChildrenLoading(true);
      apiGet('children', { accessToken })
        .then((_children) => {
          if (_children) setChildren(_children);
        })
        .finally(() => {
          setChildrenLoading(false);
          setHaveFetchedChildren(true);
        });
    }
  }, [accessToken]);

  return <Provider value={{ children: childCache }}>{childNodes}</Provider>;
};

export { DataCacheProvider };
export { Consumer as DataCacheConsumer };
export default DataCacheContext;
