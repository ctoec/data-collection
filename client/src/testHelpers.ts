import React from 'react';
import { RenderResult, render, cleanup, act } from '@testing-library/react';
import { axe } from 'jest-axe';

export const accessibilityTestHelper = (
  component: React.ReactElement,
  before?: (_: RenderResult) => Promise<void>
) => {
  it('passes AXE accessibility checks', async () => {
    await act(async () => {
      const renderResult = render(component);
      if (before) {
        await before(renderResult);
      }
      const results = await axe(renderResult.container);
      expect(results).toHaveNoViolations();
    });
    cleanup();
  });
};

export const snapshotTestHelper = (component: React.ReactElement) => {
  it('matches snapshot', async () => {
    await act(async () => {
      const { asFragment } = render(component);
      expect(asFragment()).toMatchSnapshot();
    });
  });
};
