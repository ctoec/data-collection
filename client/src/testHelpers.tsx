import React from 'react';
import { RenderResult, render, cleanup, act } from '@testing-library/react';
import { axe } from 'jest-axe';
import { BrowserRouter } from 'react-router-dom';

export const renderHelper = async (
  component: React.ReactElement,
  opts: {
    before?: (_: RenderResult) => Promise<void>;
    wrapInRouter?: boolean;
  } = {}
) => {
  const renderResult = render(
    opts.wrapInRouter ? <BrowserRouter>{component}</BrowserRouter> : component
  );

  if (opts.before) {
    await opts.before(renderResult);
  }
  return renderResult;
};

export const accessibilityTestHelper = (
  component: React.ReactElement,
  opts: {
    before?: (_: RenderResult) => Promise<void>;
    wrapInRouter?: boolean;
  } = {}
) => {
  it('passes AXE accessibility checks', async () => {
    await act(async () => {
      const renderResult = await renderHelper(component, opts);
      const results = await axe(renderResult.container);
      expect(results).toHaveNoViolations();
    });

    cleanup();
  });
};

export const snapshotTestHelper = (
  component: React.ReactElement,
  opts: {
    before?: (_: RenderResult) => Promise<void>;
    wrapInRouter?: boolean;
  } = {}
) => {
  it('matches snapshot', async () => {
    await act(async () => {
      const renderResult = await renderHelper(component, opts);
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    cleanup();
  });
};
