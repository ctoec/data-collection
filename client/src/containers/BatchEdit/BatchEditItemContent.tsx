import React, { useState, useEffect, useContext } from 'react';
import { Child, ObjectWithValidationErrors } from '../../shared/models';
import {
  StepProps,
  StepList,
  Button,
  InlineIcon,
  LoadingWrapper,
} from '@ctoec/component-library';
import { RecordFormProps } from '../../components/Forms/types';
import { listSteps } from './listSteps';
import { nameFormatter } from '../../utils/formatters';
import { useAlerts } from '../../hooks/useAlerts';
import { hasValidationErrorForField } from '../../utils/hasValidationError';
import { apiGet } from '../../utils/api';
import AuthenticationContext from '../../contexts/AuthenticationContext/AuthenticationContext';
import { Link } from 'react-router-dom';
import { getCurrentEnrollment } from '../../utils/models';
import { mutateCallback } from 'swr/dist/types';

type BatchEditItemContentProps = {
  childId: string;
  organizationId?: string;
  moveNextRecord: (_?: Child[]) => void;
  mutate?: (_: mutateCallback<Child[]>, __: boolean) => void;
};

export const BatchEditItemContent: React.FC<BatchEditItemContentProps> = ({
  childId,
  moveNextRecord,
  mutate,
}) => {
  const [child, setChild] = useState<Child>();

  const { setAlerts } = useAlerts();

  const [steps, setSteps] = useState<StepProps<RecordFormProps>[]>();
  const [activeStepKey, setActiveStep] = useState<string>();

  const { accessToken } = useContext(AuthenticationContext);
  const [triggerRefetchCount, setTriggerRefetchCount] = useState(0);

  // Reset state when new child
  useEffect(() => {
    setChild(undefined);
    setActiveStep(undefined);
    setSteps(undefined);
  }, [childId]);

  // Generate steps for the current child
  // after state is set (child has been fetched)
  useEffect(() => {
    if (child) {
      const _steps = listSteps(child);
      setSteps(_steps);
      if (_steps.length) {
        setActiveStep(_steps[0].key);
      }
    }
  }, [child?.id]);

  // Function to progress to next step
  const moveNextStep = (records?: Child[]) => {
    if (!activeStepKey || !steps) return;

    const activeStepIdx = steps.findIndex((step) => step.key === activeStepKey);
    if (activeStepIdx === steps.length - 1) {
      moveNextRecord(records);
    } else {
      setActiveStep(steps[activeStepIdx + 1].key);
    }
  };

  // Fetch child record
  // and if fetch is re-fetch (triggerRefetchCount > 0)
  // then progress to next step
  useEffect(() => {
    apiGet(`children/${childId}`, accessToken)
      .then((updatedChild) => {
        setChild(updatedChild);
        let updatedChildren: Child[] = [];
        if (mutate) {
          mutate((children: Child[]) => {
            if (children) {
              updatedChildren = [...children];
              const idx = updatedChildren.findIndex((c) => c.id === childId);
              if (idx > -1) updatedChildren.splice(idx, 1, updatedChild);
              return updatedChildren;
            }
            return children;
          }, false);
        }

        if (triggerRefetchCount) moveNextStep(updatedChildren);
      })
      .catch((err) => {
        throw new Error(err);
      });
  }, [accessToken, triggerRefetchCount, !!child]);

  const props: RecordFormProps = {
    child,
    afterSaveSuccess: () => setTriggerRefetchCount((r) => r + 1),
    setAlerts,
    hideHeader: true,
    topHeadingLevel: 'h4', // needed bc we still need the headers within the forms to be accurately nested
    showFieldOrFieldset: showFieldInBatchEditForm,
    AdditionalButton: (
      <Button
        text="Skip"
        onClick={() => moveNextRecord()}
        appearance="outline"
      />
    ),
  };

  const AllComplete = (
    <div className="margin-y-2">
      <div className="text-center font-body-xl">
        <InlineIcon icon="complete" />
      </div>
      <div className="text-center text-bold font-body-lg">
        {`${child?.firstName}'s record is now complete!`}
      </div>
    </div>
  );

  if (!child) {
    return <LoadingWrapper loading={true} />;
  }

  const currentEnrollment = getCurrentEnrollment(child);
  return (
    <>
      <div className="padding-left-2 padding-right-2 padding-bottom-3">
        <div className="display-flex flex-row flex-justify flex-align-end">
          <h2 className="margin-bottom-0">{nameFormatter(child)}</h2>
          <div className="text-baseline text-base">
            Date of birth: {child.birthdate?.format('MM/DD/YYYY') || 'Missing'}
          </div>
        </div>
        <div className="margin-top-1 display-flex flex-row flex-justify">
          <Link to={`/edit-record/${child.id}`}>View full profile</Link>
          <div className="text-base">
            {currentEnrollment?.ageGroup || 'Missing age group'} â€”{' '}
            {currentEnrollment?.site?.siteName || 'Missing site'}
          </div>
        </div>
      </div>
      <div className="padding-1 border-top-1px border-base-light">
        {steps && steps.length ? (
          <StepList<RecordFormProps>
            key={child.id}
            steps={steps}
            props={props}
            activeStep={activeStepKey || ''}
            headerLevel="h3"
          />
        ) : (
          AllComplete
        )}
      </div>
    </>
  );
};

// Function that defined what fields should be shown in
// forms during batch edit
export const showFieldInBatchEditForm = (
  formData: ObjectWithValidationErrors | undefined,
  fields: string[]
) => {
  if (!formData) return false;
  for (let i = 0; i < fields.length; i++) {
    if (
      // special case to account for separation of
      // enrollment and funding forms in batch edit flow
      // (the 'fundings' field in enrollment form shoud never be shown)
      fields[i] !== 'fundings' &&
      hasValidationErrorForField(formData, fields[i])
    ) {
      return true;
    }
  }
  return false;
};
