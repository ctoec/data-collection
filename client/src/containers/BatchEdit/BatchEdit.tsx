import React, { useContext, useState, useEffect } from 'react';
import { getH1RefForTitle } from '../../utils/getH1RefForTitle';
import {
  SideNav,
  TextWithIcon,
  Success,
  Button,
  ArrowRight,
  InlineIcon,
} from '@ctoec/component-library';
import { BackButton } from '../../components/BackButton';
import DataCacheContext from '../../contexts/DataCacheContext/DataCacheContext';
import { Child } from '../../shared/models';
import { hasValidationError } from '../../utils/hasValidationError';
import pluralize from 'pluralize';
import { nameFormatter } from '../../utils/formatters';
import { Link } from 'react-router-dom';

const BatchEdit: React.FC = () => {
  const h1Ref = getH1RefForTitle();
  const { children } = useContext(DataCacheContext);

  const [allRecords, setAllRecords] = useState<Child[]>([]);
  useEffect(() => {
    if (!children.loading) {
      setAllRecords(children.records.filter(hasValidationError));
    }
  }, [children.loading]);
  const missingInformationRecordsCount = allRecords.filter(hasValidationError)
    .length;

  const [activeRecordIdx, setActiveRecordIdx] = useState<number>();
  useEffect(() => {
    if (allRecords.length) setActiveRecordIdx(0);
  }, [allRecords.length]);

  const moveNextRecord = () => {
    // If active record is last record in the list
    if (activeRecordIdx === allRecords.length - 1) {
      // Then look for the first record that is still missing info
      const firstStillMissingInformationRecordIdx = allRecords.findIndex(
        hasValidationError
      );

      setActiveRecordIdx(
        // If a no records are missing info, then set active record to none
        firstStillMissingInformationRecordIdx < 0
          ? undefined
          : // otherwise set active record to first that is still missing info
            firstStillMissingInformationRecordIdx
      );
    }

    // otherwise, move to next record in the list
    setActiveRecordIdx((idx) => (idx === undefined ? idx : idx + 1));
  };

  return (
    <div className="grid-container">
      <BackButton text="Back to roster" location="/roster" />
      <h1 ref={h1Ref} className="margin-bottom-1">
        Add needed information
      </h1>
      {children.loading ? (
        <>Loading</>
      ) : (
        <>
          <p className="display-flex font-body-lg height-5 line-height-body-6 margin-y-0">
            {missingInformationRecordsCount ? (
              <>
                <div className="text-white text-bold text-center bg-primary width-5 radius-pill margin-right-1">
                  {missingInformationRecordsCount}
                </div>
                {pluralize('record', missingInformationRecordsCount)}{' '}
                {pluralize('has', missingInformationRecordsCount)} missing or
                incomplete info required to submit your data
              </>
            ) : (
              <>
                {/* TODO make it possible to make this icon big */}
                <InlineIcon className="height-4 width-4" icon="complete" /> All
                enrollments are complete
              </>
            )}
          </p>
          <SideNav
            externalActiveItemIndex={activeRecordIdx}
            items={allRecords.map((record) => ({
              title: (
                <span>
                  {nameFormatter(record)}
                  {!hasValidationError(record) && (
                    <InlineIcon icon="complete" />
                  )}
                </span>
              ),
              description: 'Placeholder',
              content: <div>PLACEHOLDER </div>,
            }))}
            noActiveItemContent={
              <div className="margin-x-4 margin-top-4 display-flex flex-column flex-align-center">
                <InlineIcon icon="complete" />
                <p className="font-body-lg text-bold margin-y-1">
                  All records are complete!
                </p>
                <Link to="/roster">
                  <TextWithIcon
                    text="Back to roster"
                    iconSide="right"
                    Icon={ArrowRight}
                    direction="right"
                  />
                </Link>
              </div>
            }
          />
        </>
      )}
    </div>
  );
};

export default BatchEdit;
