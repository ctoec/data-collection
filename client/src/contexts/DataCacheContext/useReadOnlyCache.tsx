import { useContext, useState, useEffect } from 'react';
import AuthenticationContext from '../AuthenticationContext/AuthenticationContext';
import { ReadOnlyDataCache } from './DataCacheContext';
import { apiGet } from '../../utils/api';

export const useReadOnlyCache = <T extends { id: any }>(apiPath: string) => {
  const { accessToken } = useContext(AuthenticationContext);
  const [records, setRecords] = useState<T[]>([]);
  const [loading, setLoading] = useState(true);
  const [haveFetched, setHaveFetched] = useState(false);

  const refetch = () => setHaveFetched(false);
  const getRecordById = (id: number | string) =>
    records.find((r) => r.id === id);

  useEffect(() => {
    if (!haveFetched && accessToken) {
      setLoading(true);
      apiGet(apiPath, accessToken)
        .then((_records) => {
          if (_records) setRecords(_records);
        })
        .finally(() => {
          setLoading(false);
          setHaveFetched(true);
        });
    }
  }, [accessToken, apiPath, haveFetched]);

  return {
    records,
    loading,
    refetch,
    getRecordById,
  } as ReadOnlyDataCache<T>;
};
