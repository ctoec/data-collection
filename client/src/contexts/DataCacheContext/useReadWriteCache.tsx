import { useState, useEffect, useContext } from 'react';
import { apiGet } from '../../utils/api';
import AuthenticationContext from '../AuthenticationContext/AuthenticationContext';
import { ReadWriteDataCache } from './DataCacheContext';

export const useReadWriteCache = <T extends { id: any }>(apiPath: string) => {
  const { accessToken } = useContext(AuthenticationContext);
  const [records, setRecords] = useState<T[]>([]);
  const [loading, setLoading] = useState(true);
  const [haveFetched, setHaveFetched] = useState(false);

  const addOrUpdateRecord = (_record: T) => {
    setRecords((existingRecords) => [
      ...existingRecords.filter((r) => r.id !== _record.id),
      _record,
    ]);
  };

  const removeRecordById = (id: string) => {
    setRecords((existingRecords) => [
      ...existingRecords.filter((r) => r.id !== id),
    ]);
  };

  const getRecordById = (id: number | string) =>
    records.find((r) => r.id === id);

  const refetch = () => setHaveFetched(false);

  useEffect(() => {
    if (!haveFetched && accessToken) {
      setLoading(true);
      apiGet(apiPath, { accessToken })
        .then((_records) => {
          if (_records) setRecords(_records);
        })
        .finally(() => {
          setLoading(false);
          setHaveFetched(true);
        });
    }
  }, [apiPath, accessToken, haveFetched]);

  return {
    records,
    loading,
    addOrUpdateRecord,
    removeRecordById,
    getRecordById,
    refetch,
  } as ReadWriteDataCache<T>;
};
