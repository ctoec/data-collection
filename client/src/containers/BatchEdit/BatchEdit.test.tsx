import React from 'react';
import BatchEdit from './BatchEdit';
import { snapshotTestHelper, accessibilityTestHelper } from '../../testHelpers';
import DataCacheContext, {
  ReadOnlyDataCache,
  ReadWriteDataCache,
} from '../../contexts/DataCacheContext/DataCacheContext';
import {
  Child,
  Family,
  FundingSpace,
  ReportingPeriod,
} from '../../shared/models';
import moment from 'moment';
import { ValidationError } from 'class-validator';

const children: Child[] = [
  {
    id: '00000000-0000-0000-0000-000000000000',
    firstName: 'First',
    lastName: 'Last',
    birthdate: moment.utc('2019-06-01', 'YYYY-MM-DD'),
    family: {} as Family,
    validationErrors: [{} as ValidationError],
  } as Child,
];

const cacheContext = {
  children: {
    records: [],
    loading: false,
    refetch: jest.fn(),
    addOrUpdateRecord: jest.fn(),
    getRecordById: jest.fn(),
    removeRecordById: jest.fn(),
  },
  fundingSpaces: {} as ReadOnlyDataCache<FundingSpace>,
  reportingPeriods: {} as ReadOnlyDataCache<ReportingPeriod>,
};

describe('BatchEdit', () => {
  snapshotTestHelper(
    <DataCacheContext.Provider value={cacheContext}>
      <BatchEdit />
    </DataCacheContext.Provider>,
    {
      wrapInRouter: true,
      name: 'matches snapshot when no records are missing info',
    }
  );

  snapshotTestHelper(
    <DataCacheContext.Provider
      value={{
        ...cacheContext,
        children: {
          records: children,
          loading: false,
          addOrUpdateRecord: jest.fn(),
          getRecordById: () => children[0],
          refetch: jest.fn(),
          removeRecordById: jest.fn(),
        },
      }}
    >
      <BatchEdit />
    </DataCacheContext.Provider>,
    {
      wrapInRouter: true,
      name: 'matches snapshot when >0 records are missing info',
    }
  );

  accessibilityTestHelper(
    <DataCacheContext.Provider value={cacheContext}>
      <BatchEdit />
    </DataCacheContext.Provider>,
    { wrapInRouter: true }
  );
});
