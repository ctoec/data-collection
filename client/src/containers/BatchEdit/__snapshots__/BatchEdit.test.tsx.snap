// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BatchEdit matches snapshot when >0 records are missing info 1`] = `
<DocumentFragment>
  <div
    class="grid-container"
  >
    <button
      class="usa-button usa-button--unstyled margin-2 text-bold"
      type="button"
    >
      <span
        class="oec-text-with-icon"
      >
        <svg
          class="oec-text-with-icon__icon oec-text-with-icon__icon--direction-left oec-text-with-icon__icon--side-left"
        >
          arrowRight.svg
        </svg>
        Back to roster
      </span>
    </button>
    <h1
      class="margin-bottom-1"
    >
      Add needed information
    </h1>
    <div
      class="oec-loading-wrapper"
    >
      <div
        aria-hidden="false"
        class=""
      >
        <div
          class="display-flex font-body-lg height-5 line-height-body-6 margin-y-0"
        >
          <span
            class="height-4 width-4 oec-inline-icon oec-inline-icon--complete"
          >
            <span
              class="usa-sr-only"
            >
              (complete)
            </span>
          </span>
           All enrollments are complete
        </div>
        <div
          class="oec-sidenav grid-row"
        >
          <div
            class="mobile-lg:grid-col-4"
          >
            <nav>
              <ul />
            </nav>
          </div>
          <div
            class="oec-sidenav__content mobile-lg:grid-col-12"
          >
            <div
              class="margin-x-4 margin-top-4 display-flex flex-column flex-align-center"
            >
              <span
                class="oec-inline-icon oec-inline-icon--complete"
              >
                <span
                  class="usa-sr-only"
                >
                  (complete)
                </span>
              </span>
              <p
                class="font-body-lg text-bold margin-y-1"
              >
                All records are complete!
              </p>
              <a
                href="/roster"
              >
                <span
                  class="oec-text-with-icon"
                >
                  Back to roster
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="oec-loading-wrapper__indicator bg-white"
        hidden=""
      >
        <div
          class="MuiCircularProgress-root indeterminate MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BatchEdit matches snapshot when no records are missing info 1`] = `
<DocumentFragment>
  <div
    class="grid-container"
  >
    <button
      class="usa-button usa-button--unstyled margin-2 text-bold"
      type="button"
    >
      <span
        class="oec-text-with-icon"
      >
        <svg
          class="oec-text-with-icon__icon oec-text-with-icon__icon--direction-left oec-text-with-icon__icon--side-left"
        >
          arrowRight.svg
        </svg>
        Back to roster
      </span>
    </button>
    <h1
      class="margin-bottom-1"
    >
      Add needed information
    </h1>
    <div
      class="oec-loading-wrapper"
    >
      <div
        aria-hidden="true"
        class="opacity-0"
      >
        <div
          class="display-flex font-body-lg height-5 line-height-body-6 margin-y-0"
        >
          <span
            class="height-4 width-4 oec-inline-icon oec-inline-icon--complete"
          >
            <span
              class="usa-sr-only"
            >
              (complete)
            </span>
          </span>
           All enrollments are complete
        </div>
        <div
          class="oec-sidenav grid-row"
        >
          <div
            class="mobile-lg:grid-col-4"
          >
            <nav>
              <ul />
            </nav>
          </div>
          <div
            class="oec-sidenav__content mobile-lg:grid-col-12"
          >
            <div
              class="margin-x-4 margin-top-4 display-flex flex-column flex-align-center"
            >
              <span
                class="oec-inline-icon oec-inline-icon--complete"
              >
                <span
                  class="usa-sr-only"
                >
                  (complete)
                </span>
              </span>
              <p
                class="font-body-lg text-bold margin-y-1"
              >
                All records are complete!
              </p>
              <a
                href="/roster"
              >
                <span
                  class="oec-text-with-icon"
                >
                  Back to roster
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="oec-loading-wrapper__indicator bg-white"
      >
        <div
          class="MuiCircularProgress-root indeterminate MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
